packages/kbn-esql-validation-autocomplete/src/validation/validation.ts

```diff
@@ -461,9 +455,9 @@ function validateFunction(
   const failingSignatures: ESQLMessage[][] = [];
   for (const signature of matchingSignatures) {
     const failingSignature: ESQLMessage[] = [];
-    signature.params.forEach((argDef, index) => {
-      const outerArg = astFunction.args[index]!;
-      if (!outerArg && argDef.optional) {
+    astFunction.args.forEach((outerArg, index) => {
+      const argDef = getParamAtPosition(signature, index);
+      if ((!outerArg && argDef?.optional) || !argDef) {
```

Comment:
If a signature is variadic (has `minParams` declared) then the `params` array can have fewer elements than actually exist in the AST. This led to some AST parameters not being validated. All better now (test generator++)